File: Anul 2/Semestrul 2/MPP/labs/Tema 5-proiect Server Csharp-Client Java/proiectServerClientJava/Networking/src/main/java/proiectServerClientJava/network/utils/ProtobuffConcurrentServer.java
package proiectServerClientJava.network.utils;


import proiectServerClientJava.network.protobuffprotocol.ProtoClientWorker;
import proiectServerClientJava.service.IService;

import java.net.Socket;

public class ProtobuffConcurrentServer extends AbsConcurrentServer{
    private IService chatServer;
    public ProtobuffConcurrentServer(int port, IService chatServer) {
        super(port);
        this.chatServer = chatServer;
        System.out.println("Chat- ChatProtobuffConcurrentServer");
    }

    @Override
    protected Thread createWorker(Socket client) {
        ProtoClientWorker worker=new ProtoClientWorker(chatServer,client);
        Thread tw=new Thread(worker);
        return tw;
    }
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 5-proiect Server Csharp-Client Java/proiectSCcsharp/angajatNetworking/utils/ProtobuffConcurrentServer.cs
using log4net;
using proiectSCcsharp.networking.utils;
using proiectSCcsharp.services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Sockets;
using System.Text;
using System.Threading.Tasks;
using Chat.Protocol;
using Google.Protobuf;
using protobuf;

namespace proiectSCcsharp.networking.utils
{
    public class ProtobuffConcurrentServer:AbsConcurrentServer
    {
        private readonly IService service;
        private static readonly ILog logger = LogManager.GetLogger("RpcConcurrentServer");

        private ProtoClientWorker worker;
        public ProtobuffConcurrentServer(string host, int port, IService service) : base(host, port)
        {
            this.service = service;
            logger.Info("ProtobuffConcurrentServer created");
            Console.WriteLine(@"ProtobuffConcurrentServer created");
        }
        protected override Thread CreateWorker(TcpClient client)
        {
            worker = new ProtoClientWorker(service, client);
            return new Thread(new ThreadStart(worker.run));
        }
    }
}
